server.js

```diff
@@ -140,22 +211,68 @@ apiRoutes.use(function(req, res, next) {
 
 //(GET http://localhost:8080/api)
 apiRoutes.get('/getTokenData', function(req, res) {
-    data = readInJson();
-    res.json({ message: 'API is working!' });   
+    readInJson.then((data)=> {
+        const fiveMinsAgo = moment().unix() - 5*60;
+        if(data.timeStamp < fiveMinsAgo){ //wants to be smalller <
+            console.log("Json updating");            
+            updateJson.then((data) =>{res.json({ data })});
+            return data
+        } else {
+            res.json({ data });  
+            return data;
+        }
+    })
 });
 
-//We don't need post
-// apiRoutes.post('/dodajkorisnika', function (req, res, next) {
+//Post for changing the value of Ether Chris has taken out

```

Comment:
This feature is probably better implemented by a call to an external service that has built-in authentication for Chris (something like Firebase). We can then hide the Firebase credentials in an env variable and it means this service stays entirely passive to clients.